etc/systemd/system/baryssh-server.service

```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=barySSH server service
+After=sshd.service
+Requires=sshd.service
```

Comment:
baryssh has `ssh' in name but can proxy pretty much any traffic. I'd skip dependency on sshd service - maybe start it after `networking'. Especially since it doesn't really try to connect to the SSH unless the user opens connection first - it's not a hard dependency. This will also ease joining of client and server parts into one.